### PR TITLE
CVE-2015-6774 and CVE-2011-3964

### PR DESCRIPTION
**Pull request by RiceOnly**
Updated the CVE-2015-6774 and CVE-2011-3964.

### CODE REVIEW

File: cves/CVE-2011-3964.yml [hunk 1]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  The drag and drop feature within Google chrome before version 17 was not properly 
+  implemented. This allowed spoofing of legitimate urls through javascript when a link was 
+  dragged into the address bar.
```
Comment (roda0313) [inline cves/CVE-2011-3964.yml]: Be more specific here, does this mean you had to drag some javascript code into the url bar or you simply dragged an actual url?

File: cves/CVE-2011-3964.yml [hunk 2]
```diff
@@ -54,8 +57,9 @@ fixes:
 - :commit: 0c14577c9905bd8161159ec7eaac810c594508d0
   :note: ''
 vccs:
-- :commit:
-  :note: 
+- :commit: 067095afacfdf4673bc397b3ca4623d037e7afc7
+  :note: This was introduced when fixing another bug:82181 where javascript code is stripped
+         before it is pasted into the omnibox.
```
Comment (roda0313) [inline cves/CVE-2011-3964.yml]: What is an omnibox? explain that its the subsystem and its function or use different wording

File: cves/CVE-2011-3964.yml [hunk 3]
```diff
@@ -197,4 +211,14 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The mistakes that were made was that after stripping the javascript schema, having to
```
Comment (roda0313) [inline cves/CVE-2011-3964.yml]: "The mistakes that was made was that..." small grammar fix in this sentence

File: cves/CVE-2015-6774.yml [hunk 4]
```diff
@@ -32,7 +32,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: | 
+  The vulnerability was found in one of the functions relating to loading time 
+  within the Extensions of Google Chrome.
```
Comment (roda0313) [inline cves/CVE-2015-6774.yml]: within what extensions? So this has nothing to do with google chrome and an extension wasn't working correctly? I don't fully understand this sentence, consider rewording

File: cves/CVE-2015-6774.yml [hunk 5]
```diff
@@ -32,7 +32,14 @@ description_instructions: |
...
+  
+  Modification of a pointer through javascript can give remote attackers the 
+  ability to cause denial of service. Specifically, when the pointer used for 
+  reporting load times data is modified by javascript, the use of unexpected
```
Comment (roda0313) [inline cves/CVE-2015-6774.yml]: So a user has to inject a javascript script into the code somehow for this to work? How do they inject this code? Explain

File: cves/CVE-2015-6774.yml [hunk 6]
```diff
@@ -196,6 +212,18 @@ mistakes:
...
+    having explicitly written code prevents some of these vulnerablities. 
+    
+    Based on the CWE entry, to mitigate such vulnerability, one should try to use a language that 
+    provides automatic memory management. This was not done because execution should be as optimal 
+    when it comes to speed and to do that, carrying out your own memory management gives more flexibility.
+    For the implementation fix, the programmer did set the pointer to null before carrying out further
+    implementation of the pointer.
+
+
+
```
Comment (roda0313) [inline cves/CVE-2015-6774.yml]: 2 Upvotes

File: cves/CVE-2011-3964.yml [hunk 7]
```diff
@@ -197,4 +211,14 @@ mistakes:
...
+    The mistakes that were made was that after stripping the javascript schema, having to 
+    not navigate to the link allowed for social engineering to take place and spoofing of 
+    websites to occur. It was more of a design mistake as the use case of dragging and dropping
+    a link to url address bar was not carried out fully to the end. You would expect the browser
+    to navigate to the link and not stop. 
+    
+    Based on the CWE entry for this vulnerability, they are not exactly carrying out an architecture
+    and design mitigation of warning the user that they are leaving the page if there is javascript ran 
+    behind the scenes. But they are sanitizing the javascript and navigating to the url to prevent spoofing 
+    of the real website.
```
Comment (roda0313) [inline cves/CVE-2011-3964.yml]: 2 Upvotes

Comment (RobertHofmann) [reply cves/CVE-2011-3964.yml]: 3 upvotes

File: cves/CVE-2015-6774.yml (hunk 6 above)
Comment (RobertHofmann) [reply cves/CVE-2015-6774.yml]: 2 upvotes

File: cves/CVE-2015-6774.yml [hunk 8]
```diff
@@ -32,7 +32,14 @@ description_instructions: |
...
   keep too.
-description: 
+description: | 
+  The vulnerability was found in one of the functions relating to loading time 
+  within the Extensions of Google Chrome.
+  
+  Modification of a pointer through javascript can give remote attackers the 
+  ability to cause denial of service. Specifically, when the pointer used for 
+  reporting load times data is modified by javascript, the use of unexpected 
+  values can occur.
```
Comment (RobertHofmann) [inline cves/CVE-2015-6774.yml]: The last sentence should be clarified. What happens in the case where an unexpected value is used? Is this exclusively a repudiation issue?

File: cves/CVE-2015-6774.yml [hunk 9]
```diff
@@ -166,8 +179,11 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      It applies here because you want to ensure that the variables coming in
+      are not what you expect them to be and explicitly check for that.Setting a
```
Comment (RobertHofmann) [inline cves/CVE-2015-6774.yml]: that.Setting needs a space

File: cves/CVE-2015-6774.yml [hunk 10]
```diff
@@ -196,6 +212,18 @@ mistakes:
...
+    every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The mistake made was assuming the type of the input before carrying out execution of
+    the variable. When working in a programming language that has the programmer handling memory,
+    having explicitly written code prevents some of these vulnerablities. 
+    
+    Based on the CWE entry, to mitigate such vulnerability, one should try to use a language that 
+    provides automatic memory management. This was not done because execution should be as optimal
```
Comment (RobertHofmann) [inline cves/CVE-2015-6774.yml]: This second sentence looks improper. Maybe change "be as optimal when" to "be optimal when" or "be as optimal as possible when".

File: cves/CVE-2015-6774.yml [hunk 11]
```diff
@@ -196,6 +212,18 @@ mistakes:
...
+  answer: |
+    The mistake made was assuming the type of the input before carrying out execution of
+    the variable. When working in a programming language that has the programmer handling memory,
+    having explicitly written code prevents some of these vulnerablities. 
+    
+    Based on the CWE entry, to mitigate such vulnerability, one should try to use a language that 
+    provides automatic memory management. This was not done because execution should be as optimal 
+    when it comes to speed and to do that, carrying out your own memory management gives more flexibility.
+    For the implementation fix, the programmer did set the pointer to null before carrying out further
+    implementation of the pointer.
```
Comment (RobertHofmann) [inline cves/CVE-2015-6774.yml]: Would you consider this a proper fix? There is no indication of whether or not it is.

**Pull request merged**